### PR TITLE
Fixing material builder issues

### PR DESCRIPTION
**Pull request by santorac**
Fixed a simple reversed logic error in MaterialAssetBuilderComponent that caused shader file changes to trigger FBX asset rebuilds.

Made material and material type file builders depend on image files. This fixes a particular edge case: image source file is missing, the material fails to process, the image file appears later, and the material builder does not wake up to try again. (Actually it seems this issue could have been occurring for other file types as well because AddPossibleDependencies() was not adding source dependencies, which are required for waking up when a missing file appears).

Tested .material file...
- Shut down the AP
- Rename Gems\Atom\TestData\TestData\Textures\checker8x8_gray_512.png to something else
- Add some whitespace in  Gems\Atom\TestData\TestData\Materials\StandardPbrTestCases\004_MetalMap.material (this will trigger the material to rebuild when the AP is run)
- Start the AP
- 004_MetalMap reports a warning about the missing image.
- Rename the image back to checker8x8_gray_512.png
- The material reprocesses and clears the warningn.

Tested .materialtype file...
- Shut down the AP
- Rename Gems\Atom\Feature\Common\Assets\Textures\DefaultBlendMask_layers.png to something else
- Add some whitespace in Gems\Atom\Feature\Common\Assets\Materials\Types\StandardMultilayerPBR.materialtype (this will trigger the material to rebuild when the AP is run)
- Start the AP
- StandardMultilayerPBR fails to build because it cannot find the image.
- Rename the image back to DefaultBlendMask_layers.png
- The material type reprocesses and succeeds.

Tested editing the content of DefaultBlendMask_layers.png and verified that this does not cause StandardMultilayerPBR.materialtype to reprocess (because of the OrderOnce job dependency).

### CODE REVIEW

File: Gems/Atom/RPI/Code/Source/RPI.Edit/Material/MaterialSourceData.cpp [hunk 1]
```diff
@@ -365,6 +365,13 @@ namespace AZ
 
             return Failure();
         }
+        
+        /*static*/ bool MaterialSourceData::LooksLikeImageFileReference(const MaterialPropertyValue& value)
```
Comment (jeremyong-az) [inline Gems/Atom/RPI/Code/Source/RPI.Edit/Material/MaterialSourceData.cpp:369]: What's the `/*static*/` comment for?

**Review by jeremyong-az — APPROVED** [review]

**Review by antonmic — APPROVED** [review]

File: Gems/Atom/RPI/Code/Source/RPI.Edit/Material/MaterialSourceData.cpp (hunk 1 above)
Comment (santorac) [reply Gems/Atom/RPI/Code/Source/RPI.Edit/Material/MaterialSourceData.cpp:369]: Copy/paste from the header. My personal style is to keep the static keyword commented out for clarity.

Comment (santorac) [reply Gems/Atom/RPI/Code/Source/RPI.Edit/Material/MaterialSourceData.cpp:369]: Actually, if you search the codebase for "/*static*/" you'll see I'm not the only one who likes this.

**Comment by jeremyong-az** [comment]
FYI @santorac weird test failure:

```
[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-8056/2/pipeline#step-613-log-1204)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-8056/2/pipeline#step-613-log-1205)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-8056/2/pipeline#step-613-log-1206)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-8056/2/pipeline#step-613-log-1207)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-8056/2/pipeline#step-613-log-1208)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-8056/2/pipeline#step-613-log-1209)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-8056/2/pipeline#step-613-log-1210)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-8056/2/pipeline#step-613-log-1211)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-8056/2/pipeline#step-613-log-1212)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-8056/2/pipeline#step-613-log-1213)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-8056/2/pipeline#step-613-log-1214)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-8056/2/pipeline#step-613-log-1215)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-8056/2/pipeline#step-613-log-1216)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-8056/2/pipeline#step-613-log-1217)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-8056/2/pipeline#step-613-log-1218)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-8056/2/pipeline#step-613-log-1219)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-8056/2/pipeline#step-613-log-1220)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-8056/2/pipeline#step-613-log-1221)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-8056/2/pipeline#step-613-log-1222)[2022-03-04T19:40:00.528Z] Python: RedirectOutput installedPython: [ERROR] o3de.manifest: Manifest json failed to load at path "C:\cygwin\home\lybuilder\.o3de\o3de_manifest.json": Expecting value: line 1 column 1 (char 0)

[2022-03-04T19:40:00.528Z] PythonBindings: 

[2022-03-04T19:40:00.528Z] ==================================================================

[2022-03-04T19:40:00.528Z] PythonBindings: Trace::Warning

[2022-03-04T19:40:00.528Z]  D:/workspace/o3de/Code/Tools/ProjectManager/Source/PythonBindings.cpp(366): 'class AZ::Outcome<void,class AZStd::basic_string<char,struct AZStd::char_traits<char>,class AZStd::allocator> > __cdecl O3DE::ProjectManager::PythonBindings::ExecuteWithLockErrorHandling(class AZStd::function<void __cdecl(void)>)'

[2022-03-04T19:40:00.528Z] PythonBindings: Python exception JSONDecodeError: Expecting value: line 1 column 1 (char 0)

[2022-03-04T19:40:00.528Z] 

[2022-03-04T19:40:00.528Z] At:

[2022-03-04T19:40:00.528Z]   D:\workspace\o3de\python\runtime\python-3.7.12-rev1-windows\python\lib\json\decoder.py(355): raw_decode

[2022-03-04T19:40:00.528Z]   D:\workspace\o3de\python\runtime\python-3.7.12-rev1-windows\python\lib\json\decoder.py(337): decode

[2022-03-04T19:40:00.528Z]   D:\workspace\o3de\python\runtime\python-3.7.12-rev1-windows\python\lib\json\__init__.py(348): loads

[2022-03-04T19:40:00.528Z]   D:\workspace\o3de\python\runtime\python-3.7.12-rev1-windows\python\lib\json\__init__.py(296): load

[2022-03-04T19:40:00.528Z]   d:\workspace\o3de\scripts\o3de\o3de\manifest.py(187): load_o3de_manifest

[2022-03-04T19:40:00.528Z]   d:\workspace\o3de\scripts\o3de\o3de\manifest.py(593): get_registered

[2022-03-04T19:40:00.528Z]   d:\workspace\o3de\scripts\o3de\o3de\engine_template.py(1514): create_project

[2022-03-04T19:40:00.528Z] 

[2022-03-04T19:40:00.528Z] PythonBindings: ==================================================================

[2022-03-04T19:40:00.528Z] D:/workspace/o3de/Code/Tools/ProjectManager/tests/PythonBindingsTests.cpp(64): error: Value of: result.IsSuccess()

[2022-03-04T19:40:00.528Z]   Actual: false

[2022-03-04T19:40:00.528Z] Expected: true
```

**Pull request merged**